doc/limit/SONiC Limit user session.md

```diff
@@ -0,0 +1,222 @@
+  * [1.1 Limit the login session per user/group/system](#11-limit-the-login-session-per-user/group/system)
+  * [1.3 Default limitation](#13-default-limitation)
+- [2 Configuration and Management Requirements](#2-configuration-and-management-requirements)
+  * [2.1 SONiC CLI](#21-sonic-cli)
+  * [2.2 Config DB](#22-config-db)
+- [3 Design](#design)
+  * [3.1 Login Limit Implementation](#31-login-limit-implementation)
+  * [3.2 Default login session limitation Implementation](#32-default-login-session-limitation-Implementation)
+  * [3.3 ConfigDB Schema](#33-configdb-schema)
+  * [3.4 CLI](#34-cli)
+- [4 Error handling](#error-handling)
+- [5 Serviceability and Debug](#serviceability-and-debug)
+- [6 Unit Test](#unit-test)
+  * [6.1 Login session enable/disable test](#61-login-session-enable-disable-test)
+  * [6.2 Max login session test](#62-max-login-session-test)
+- [8 References](#references)
+
+
+# About this Manual
+This document provides a detailed description on the new features for:
+ - Limit the number of logins per user/group/system.
+ - Default limit user login session by device information.
+ - Design a scalable framework of config command and ConfigDB to support more resource type, for example limit user CPU/memory by cgroup and systemd user.slices.
+
+## SONiC memory issue sloved by this feature.
+ - Currenly SONiC enabled OOM killer, and set 2 to /proc/sys/vm/panic_on_oom, which will trigger kernel panic when OOM. This is by design to protect SONiC key process and container.
+ - A typical switch device have 4 GB memory and sonic usually will use 1.5 GB for dockers, and 500 MB for system process. so there will be 2 GB free memory for user. also sonic not enable swap for most device.
+ - When user run some command trigger OOM, SONiC will kernel panic. for example:
+   - Multiple user login to device, some service may create 10+ concurrent sesstion login to device.
+   - Some user script/command take too much memory, currently 'show' command will take 60 MB memory.
+
+# 1 Functional Requirement
+## 1.1 Limit the login session per user/group/system
+ - Can set max login session count per user/group/system.
+ - When exceed maximum login count, login failed with error message.
+
+## 1.2 Default limitation
+- Default login session by device hardware and software information.
+- For customer, they may have pipelines to initialize device configuration, because this feature add new commands, the pipeline may need update. The default limitation is designed to cover most case to minimize the customer side change.
+
+# 2 Configuration and Management Requirements
+## 2.1 SONiC CLI
+ - Manage login session or memory  limit settings
+```
+    config limit login { enable | disable }
+    config limit login { add | del } {user | group | global} <name> <number>
+```
+ - Show limit
+```
+    show limit login
+```
+
+## 2.2 Config DB
+ - Login limit are fully configurable by config DB.
+
+# 3 Design
+ - Design diagram:
+
+```mermaid
```

Comment:
>mermaid [](http://example.com/codeflow?start=3&length=7)

I see an error. I am using "GitHub + Mermaid" extension. Is it a compatibility issue in the syntax?

```
Parse error on line 3:
... setting --> CONFDB[(Config DB)];%% H
-----------------------^
Expecting 'SPACE', 'GRAPH', 'DIR', 'TAGEND', 'TAGSTART', 'UP', 'DOWN', 'subgraph', 'end', 'MINUS', '--', '==', 'STR', 'STYLE', 'LINKSTYLE', 'CLASSDEF', 'CLASS', 'CLICK', 'DEFAULT', 'NUM', 'PCT', 'COMMA', 'ALPHA', 'COLON', 'BRKT', 'DOT', 'PUNCTUATION', 'UNICODE_TEXT', 'PLUS', 'EQUALS', 'MULT', got 'PS'
```

Reply:
I think it's a compatibility issue. Will improve document by upload picture.

Reply:
Fixed by change to .png